### ISSUE TITLE
Supervised integration on a subset of reference cells

### ISSUE THREAD

**Issue by krejciadam** [satijalab/seurat#3189]
Dear Seurat team, 

I have a rather conceptual question about integration: Is it somehow possible to include extra information to the integration process in a supervised way, marking populations of reference cells from the initial datasets that are expected to have identical transcriptomes? 

 A minimal example is something like this: Imagine I have 2 datasets, say from cell-line cells sequenced on 2 different technologies. Each of the 2 datasets contains 2 cell populations: 

dataset 1: unstimulated cells + cells stimulated with compound A
datset 2: unstimulated cells + cells stimulated with compound B

Say I know which cells are stimulated and which unstimulated in each of the datasets, because I've used cell hashing to label them. Now I was wondering whether it would be possible to increase the quality of integration results by including this information. Something like basing the integration only on the unstimulated cells, because I know these 2 sub-populations should align in the final integrated datasets, whereas the cells stimulated with 2 different compounds might possibly not align at all. 

Thank you!

**Comment by timoast** [comment]
This is not currently possible (Seurat 3.1.5) but we're working on some enhanced anchor filtration/exploration tools that should be available soon

**Comment by krejciadam** [comment]
Thanks for your reply, I'll stay tuned then!

**Comment by cohenp05** [comment]
Hello, 
I'm wondering if any of this functionality has been implemented in v4.0? I have a dataset of cells sequenced and prepped in parallel in different scRNAseq libraries and within those samples, some cells are virally infected or not. The viral infection changes the cellular transcriptome of course, but we want to be able to compare infection within cell types and across technologies (sort of a double integration: one integration across tech, and one integration across infection status). 

Is it kosher to integrate the object twice sequentially? Or to separate the seurat object into technology-infection state samples and then integrate all together? I've tried to regress the infection status out with SCTransform but the transcriptional changes seem to be too drastic to regress out completely. Thank you for your help!
